tests/components/zeroconf/test_usage.py

```diff
@@ -0,0 +1,56 @@
+"""Test Zeroconf multiple instance protection."""
+import zeroconf
+
+from homeassistant.components.zeroconf import async_get_instance
+from homeassistant.components.zeroconf.usage import install_multiple_zeroconf_catcher
+
+from tests.async_mock import Mock, patch
+
+
+async def test_multiple_zeroconf_instances(hass, mock_zeroconf, caplog):
+    """Test creating multiple zeroconf throws without an integration."""
+
+    zeroconf_instance = await async_get_instance(hass)
+
+    install_multiple_zeroconf_catcher(zeroconf_instance)
+
+    new_zeroconf_instance = zeroconf.Zeroconf()
+    assert new_zeroconf_instance == zeroconf_instance
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)